io-engine/src/lvs/lvs_lvol.rs

```diff
@@ -976,6 +1005,14 @@ impl LvsLvol for Lvol {
             }
             PropValue::EntityId(id) => {
                 let name = PropName::from(&prop).to_string().into_cstring();
+                // Return Ok early if the current value is already same as the
+                // new value.
+                if let Ok(pval) = self.get(PropName::EntityId).await {
+                    match pval {
+                        PropValue::EntityId(e) if e == id => return Ok(()),
+                        _ => (),
+                    }
+                };
```

Comment:
wonder if a more generic check could be done by moving the duplicated calls to set_xattr to somewhere below and compare the raw values.

Reply:
For removing duplication, IIRC Abhilash created a ticket while working on EntityId property for stats.